### PR TITLE
Keras 3 migration guide

### PR DESCRIPTION
**Pull request by divyashreepathihalli**
Moving PR from here - https://github.com/keras-team/keras-io/pull/1565

### CODE REVIEW

File: guides/migrate_to_keras_3.py [hunk 1]
```diff
@@ -0,0 +1,963 @@
...
+input_1 = {
+    "foo": keras.Input(shape=(1,), name="foo"),
+    "bar": {},
+}
+baz = None
+layer(input_1, input_2=baz)
+
+
+"""
+### Random seed behavior changes between Keras 2 and Keras 3.
```
Comment (fchollet) [inline guides/migrate_to_keras_3.py:558]: I'll touch up this section a bit after merging.

File: guides/migrate_to_keras_3.py [hunk 2]
```diff
@@ -0,0 +1,963 @@
...
+TensorFlow backend} \\
+\hline
+\end{array}
+"""
+
+"""
+## Additional developer guides
+
+This wraps up the migration guide overview. We hope that this guide has been helpful in
+successfully transitioning your code from TensorFlow to Keras 3.0. Explore a variety of
```
Comment (fchollet) [inline guides/migrate_to_keras_3.py:941]: From Keras 2 to Keras 3

**Review by fchollet — COMMENTED** [review]
Thanks for the updates!

Please only include the `.py` file in the PR next time. We'll merge and make changes in post.

File: guides/md/migrate_to_keras_3.md [hunk 3]
```diff
@@ -0,0 +1,1043 @@
...
+
+First, lets install `keras-nightly`.
+
+This example uses the TensorFlow backend (`os.environ["KERAS_BACKEND"] = "tensorflow"`).
+After you've migrated your code, you can change the "tensorflow" string to "torch"
+and click "Restart runtime", and your code will run on the PyTorch backend!
+
+
+```python
+!pip install keras-nightly
```
[the remark this answers is not in the document]
Comment (divyashreepathihalli) [reply guides/md/migrate_to_keras_3.md]: done

File: guides/md/migrate_to_keras_3.md [hunk 4]
```diff
@@ -0,0 +1,1043 @@
...
+The following snippet of code will reproduce the above error:
+
+```python
+sequential_model = keras.Sequential([
+    keras.layers.Dense(2)
+])
+sequential_model.save("saved_model")
+```
+
+How you fix it:
```
[the remark this answers is not in the document]
Comment (divyashreepathihalli) [reply guides/md/migrate_to_keras_3.md]: done

File: guides/migrate_to_keras_3.py (hunk 1 above)
Comment (divyashreepathihalli) [reply guides/migrate_to_keras_3.py:541]: SG!!

**Review by fchollet — APPROVED** [review]
Thanks for the updates! I'll merge now and we'll keep editing in post.

**Pull request merged**